### PR TITLE
Improve canvas layer index documentation

### LINKED ISSUE #81257: CanvasLayer Child Controls Z-Index Layering Issue
### Godot version

4.0.3

### System information

Windows -Godot 4.0.3

### Issue description

I need to change the rendering order of controls within the same Layer, but when I use two CanvasLayer nodes with the same Layer and place controls inside them, changing the z-index of child components does not affect their rendering order. I believe this may be incorrect; within the same layer, I should be able to change a control's z-index to alter its rendering order.

### Steps to reproduce

_% D 8 DH}{`I5I$)7WTMPG
%W0R{~VLQZBKB508}EL4A T

### Minimal reproduction project

[demo.zip](https://github.com/godotengine/godot/files/12504514/demo.zip)

### PR DESCRIPTION
**Pull request by Sauermann**
Document special case of identical canvas layer index.

resolve #81257
related to #81262

### CODE REVIEW

File: doc/classes/CanvasLayer.xml [hunk 1]
```diff
@@ -53,6 +53,7 @@
 		</member>
 		<member name="layer" type="int" setter="set_layer" getter="get_layer" default="1">
 			Layer index for draw order. Lower values are drawn behind higher values.
+			[b]Note:[/b] If multiple CanvasLayers have the same layer index, [CanvasItem]-children of one CanvasLayer are drawn behind the [CanvasItem]-children of the other CanvasLayer. Which CanvasLayer is drawn in front is non-deterministic.
```
Comment (clayjohn) [inline doc/classes/CanvasLayer.xml:56]: Is it really non-deterministic, or is it in scene tree order of the CanvasLayers?

Comment (Sauermann) [reply doc/classes/CanvasLayer.xml:56]: It is not only the scene tree order. You can see that in the Video of the linked issue.

[the remark this answers is not in the document]
Comment (Sauermann) [reply doc/classes/CanvasLayer.xml]: Thanks, I have adjusted the PR.

Comment (lawnjelly) [reply doc/classes/CanvasLayer.xml]: See also #69952 which made canvas layer ordering deterministic in 3.x.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**